react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { numberOfClicks: 0 };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(state => ({
+      numberOfClicks: ++state.numberOfClicks
```

Comment:
In react, we never assign directly to state properties. Instead, calculate a new number to use without an assignment back to `state`.